--- ./t/00-constants.t.tv	2005-07-08 04:41:06.000000000 +0200
+++ ./t/00-constants.t	2005-07-08 04:40:57.000000000 +0200
@@ -3,6 +3,7 @@
 use Getopt::Popt qw(:all);
 use strict;
 use Test;
+use Config;
 BEGIN { plan tests => 19 };
 
 # test constants: borrowed from popt.h,the constants will
@@ -18,7 +19,7 @@
 ok(POPT_ARG_INTL_DOMAIN,6);
 ok(POPT_ARG_VAL,7);
 ok((int(POPT_ARG_MASK) ^ 0x0000FFFF) == 0);
-ok((int(POPT_ARGFLAG_ONEDASH) ^ 0x80000000 ) == 0);
+ok((int(POPT_ARGFLAG_ONEDASH) ^ 0x80000000 ) == ($Config{use64bitint} eq 'define' ? 18446744069414584320 : 0));
 ok((int(POPT_ARGFLAG_DOC_HIDDEN) ^ 0x40000000 ) == 0);
 ok((int(POPT_ARGFLAG_STRIP) ^ 0x20000000 ) == 0);
 ok((int(POPT_ARGFLAG_LOGICALOPS) ^
